src/lib.rs

```diff
@@ -102,50 +104,20 @@ mod parser;
 const VERSION: &str = "0.0.1";
 const LANG: &str = "rust";
 
-#[doc(hidden)]
-pub trait ConnectionState: private::Sealed {}
-mod private {
-    pub trait Sealed {}
-    impl Sealed for super::NotConnected {}
-    impl Sealed for super::Authenticated {}
-    impl Sealed for super::Connected {}
-    impl Sealed for dyn super::OptionState {}
-}
-
-impl ConnectionState for NotConnected {}
-impl ConnectionState for Authenticated {}
-impl ConnectionState for Connected {}
```

Comment:
Sealed traits are considered to be a bit of an antipattern, unlike in the scala community where it seems they are being imported from. Leaning heavily into typestate tends to confuse contributors more than the mind-bending confusion and derailment that it can cause.

Reply:
Understood, but trying to optimize the end user experience and lean in on compiler support similar to serde. We could return different types like I did with SubscriptionHandler, but I liked the way docs were presented that were nice and clean and one struct, Connection.